sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/implementation/Utility.java

```diff
@@ -388,86 +399,89 @@ public static ExtractKeyPhrasesResultCollection toExtractKeyPhrasesResultCollect
-                        ).collect(Collectors.toList())
-                    ).orElse(new ArrayList<>());
-
+                final AnalyzeHealthcareEntitiesResult analyzeHealthcareEntitiesResult =
+                    new AnalyzeHealthcareEntitiesResult(
+                        documentEntities.getId(),
+                        documentEntities.getStatistics() == null ? null
+                            : toTextDocumentStatistics(documentEntities.getStatistics()),
+                        null);
+                // Warnings
+                final List<TextAnalyticsWarning> warnings = documentEntities.getWarnings().stream().map(
+                    textAnalyticsWarning -> new TextAnalyticsWarning(
+                        Optional.ofNullable(textAnalyticsWarning.getCode())
+                            .map(warningCodeValue -> WarningCode.fromString(warningCodeValue.toString()))
+                            .orElse(null),
+                        textAnalyticsWarning.getMessage())
+                ).collect(Collectors.toList());
+                AnalyzeHealthcareEntitiesResultPropertiesHelper.setWarnings(analyzeHealthcareEntitiesResult,
+                    IterableStream.of(warnings));
+                // Healthcare entities
                 final List<HealthcareEntity> healthcareEntities = documentEntities.getEntities().stream().map(
                     entity -> {
                         final HealthcareEntity healthcareEntity = new HealthcareEntity();
                         HealthcareEntityPropertiesHelper.setText(healthcareEntity, entity.getText());
+                        HealthcareEntityPropertiesHelper.setNormalizedText(healthcareEntity, entity.getName());
                         HealthcareEntityPropertiesHelper.setCategory(healthcareEntity, entity.getCategory());
                         HealthcareEntityPropertiesHelper.setConfidenceScore(healthcareEntity,
                             entity.getConfidenceScore());
                         HealthcareEntityPropertiesHelper.setOffset(healthcareEntity, entity.getOffset());
                         HealthcareEntityPropertiesHelper.setLength(healthcareEntity, entity.getLength());
+                        final List<EntityDataSource> entityDataSources =
+                            Optional.ofNullable(entity.getLinks()).map(
+                                links -> links.stream().map(
+                                    link -> {
+                                        final EntityDataSource dataSource = new EntityDataSource();
+                                        EntityDataSourcePropertiesHelper.setName(dataSource, link.getDataSource());
+                                        EntityDataSourcePropertiesHelper.setEntityId(dataSource, link.getId());
+                                        return dataSource;
+                                    }
+                                ).collect(Collectors.toList()))
+                                .orElse(new ArrayList<>());
```

Comment:
Can you confirm if we want to return this list as empty ot null?

Reply:
it can be null. https://github.com/Azure/azure-rest-api-specs/blob/master/specification/cognitiveservices/data-plane/TextAnalytics/preview/v3.1-preview.4/TextAnalytics.json#L1808

it is not required. Let me confirm with v-team on what they returns

Reply:
Generally, we return an empty list to client user. It's not an explicit guideline but generally makes code more convenient. 

.NET, Python is doing it as well .